### ISSUE TITLE
mount dies when large amount of large files are poured into it

### ISSUE THREAD

**Issue by maltokyo** [astrada/google-drive-ocamlfuse#568]
Hi Alessandro @astrada 
I have similar strange behaviour like in
https://github.com/astrada/google-drive-ocamlfuse/issues/538

I am trying to sync 700 files, about 132MB each (bitcoin blockchain).  Using rsyc goes well at first, but then google-drive-ocamlfuse dies, and just stops, and the mount point is not accessible anymore.  Rsync continues copying files though, I am not sure where it is copying them to.
Please could you help me, I am ripping my hair out here..
Thank you so much

======
Log: sent to you just now by email.

The command I am using to copy: rsync -av /var/lib/docker/volumes/generated_bitcoin_datadir/_data/blocks /mnt/google-drive-btcpayserver/ --progress --inplace

System Ubuntu 18.04, using latest google-drive-ocamlfuse version.  I have 24GB ram, so if you can think of a better config setup to use more RAM to solve the issue, please let me know.

Config is below, you can see what settings I have tried to attempt to solve it:
cat ~/.gdfuse/google-drive-btcpayserver/config

acknowledge_abuse=false
apps_script_format=desktop
apps_script_icon=
cache_directory=/mnt/google-drive-btcpayserver-cache
client_id=<REMOVED>
client_secret=<REMOVED>
connect_timeout_ms=50000
curl_debug_off=false
data_directory=
delete_forever_in_trash_folder=true
desktop_entry_exec=
disable_trash=true
docs_file_extension=true
document_format=desktop
document_icon=
download_docs=false
drawing_format=desktop
drawing_icon=
form_format=desktop
form_icon=
fusion_table_format=desktop
fusion_table_icon=
keep_duplicates=false
large_file_read_only=false
large_file_threshold_mb=32
log_directory=
lost_and_found=false
low_speed_limit=0
low_speed_time=0
map_format=desktop
map_icon=
max_cache_size_mb=102400
max_download_speed=0
max_memory_cache_size=1048576000
max_retries=16
max_upload_chunk_size=1099511627776
max_upload_speed=0
memory_buffer_size=1048576
metadata_cache_time=3600
metadata_memory_cache=true
metadata_memory_cache_saving_interval=60
presentation_format=desktop
presentation_icon=
read_ahead_buffers=9
read_only=false
root_folder=1kf3h-LzjEqb7z91gBtCg7JtojcZw8z5L
shared_with_me=false
spreadsheet_format=desktop
spreadsheet_icon=
sqlite3_busy_timeout=50000
stream_large_files=true
team_drive_id=
umask=0o022
verification_code=
write_buffers=true

**Comment by maltokyo** [comment]
Tried again now, and after crash, when I try to cd to the mountpoint, I get error "transport endpoint is not connected", and this is the end of the debug log:

```
[1248.205027] TID=61782: BEGIN: Flushing memory buffer (remote id=1WYxetPqCnwvQik-LkSfjUv0J2KxPegNp, content_path=/mnt/google-drive-btcpayserver-cache/1WYxetPqCnwvQik-LkSfjUv0J2KxPegNp, block_index=8)
[1248.214342] TID=61782: END: Flushing memory buffer (remote id=1WYxetPqCnwvQik-LkSfjUv0J2KxPegNp, content_path=/mnt/google-drive-btcpayserver-cache/1WYxetPqCnwvQik-LkSfjUv0J2KxPegNp, block_index=8)
[1248.214482] TID=61782: Releasing memory buffer (remote id=1WYxetPqCnwvQik-LkSfjUv0J2KxPegNp, index=8, buffer id=5, state=Dirty)
[1248.214521] TID=61782: BEGIN: Flushing memory buffer (remote id=1WYxetPqCnwvQik-LkSfjUv0J2KxPegNp, content_path=/mnt/google-drive-btcpayserver-cache/1WYxetPqCnwvQik-LkSfjUv0J2KxPegNp, block_index=7)
[1248.220553] TID=61782: Exception:Invalid_argument("Bigarray.sub: bad sub-array")
Backtrace:
[1248.220877] TID=61783: release /blocks/blk00211.dat O_WRONLY
[1248.220912] TID=61783: BEGIN: Loading resource /blocks/blk00211.dat (trashed=false) from db
[1248.221083] TID=61783: END: Loading resource /blocks/blk00211.dat (trashed=false) from db: Found (id=2069, state=Uploading)
[1321.825563] TID=2: Flushing DB...done
[2818.522064] TID=61784: getattr /
[2818.523683] TID=61784: BEGIN: Getting metadata from context
[2818.523701] TID=61784: END: Getting metadata: Valid
[2818.523709] TID=61784: BEGIN: Loading resource / (trashed=false) from db
[2818.523769] TID=61784: END: Loading resource / (trashed=false) from db: Found (id=1, state=ToDownload)
[2818.524172] TID=61785: statfs /
[2818.524207] TID=61785: BEGIN: Getting metadata from context
[2818.524215] TID=61785: END: Getting metadata: Valid
[2829.042646] TID=61786: getattr /
[2829.042746] TID=61786: BEGIN: Getting metadata from context
[2829.042765] TID=61786: END: Getting metadata: Valid
[2829.042776] TID=61786: BEGIN: Loading resource / (trashed=false) from db
[2829.042857] TID=61786: END: Loading resource / (trashed=false) from db: Found (id=1, state=ToDownload)
[2829.043291] TID=61787: statfs /
[2829.043347] TID=61787: BEGIN: Getting metadata from context
[2829.043357] TID=61787: END: Getting metadata: Valid
[2830.043130] TID=61788: statfs /
[2830.043195] TID=61788: BEGIN: Getting metadata from context
[2830.043204] TID=61788: END: Getting metadata: Valid
```

**Comment by astrada** [comment]
Thanks for reporting this issue. It's a bug caused by `write_buffers`. I will look into it as soon as I get back from vacation. As a workaround try setting that option to `false`.

**Comment by maltokyo** [comment]
Thank you, retrying now with that "false".

**Comment by maltokyo** [comment]
@astrada indeed that was the issue, when I changed to false, the error did not come back. Thank you.

**Comment by astrada** [comment]
Should be fixed in version 0.7.9.
